Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -404,10 +404,18 @@ extension InternalRouter where Self: Router {
 extension Array where Element: MapboxDirections.Route {
     func index(mostSimilarTo route: Route) -> Int? {
         let target = route.description
-        return enumerated().min { (left, right) -> Bool in
-            let leftDistance = left.element.description.minimumEditDistance(to: target)
-            let rightDistance = right.element.description.minimumEditDistance(to: target)
-            return leftDistance < rightDistance
-        }?.offset
+        
+        guard let bestCandidate = map({
+            (route: $0, editDistance: $0.description.minimumEditDistance(to: target))
+        }).enumerated().min(by: { $0.element.editDistance < $1.element.editDistance }) else { return nil }
+
+
+        // If the most similar route is still more than 50% different from the original route,
+        // we fallback to the fastest route which index is 0.
+        guard bestCandidate.element.route.description.count > 0 else { return 0 }
+        let similarityScore = Double(bestCandidate.element.editDistance) / Double(bestCandidate.element.route.description.count)
```

Comment:
`bestCandidate.element.route.description` or `target`?

Reply:
Good question, I don’t have a good intuition for this. Can we try out a few scenarios to see which would be more correct?

Reply:
If the best candidate is when we remove letters from the `target`, then we might get more than 1 here. I.e.
```
target = "AAAAAA"
bestCandidate = "AA"
editDistance = 4 // remove any 4 letters
score = 4 / len("AA") = 4 / 2 = 2
```
At the same time, if the best candidate is when you add letters, then we can get the same "larger than 1" if compared to the `target` length. 

We can divide by the sum of the sizes to guarantee that we get the value within `0...1`, but it's a bit more difficult to interpret such a score, i.e. 
```
"AAAAAA" -> "AAAAAA" = 0 / 12 = 0.00
"AAAAAA" -> "AAAAA"  = 1 / 11 = 0.09
"AAAAAA" -> "AAAA"   = 2 / 10 = 0.20
"AAAAAA" -> "AAA"    = 3 /  9 = 0.33
"AAAAAA" -> "AA"     = 4 /  8 = 0.50
"AAAAAA" -> "A"      = 5 /  7 = 0.71
"AAAAAA" -> ""       = 6 /  6 = 1.00

"AAAAAA" -> "AAAABB" = 2 / 12 = 0.17
"AAAAAA" -> "AAABBB" = 3 / 12 = 0.25
"AAAAAA" -> "AABBBB" = 4 / 12 = 0.33
"AAAAAA" -> "BBBBBB" = 6 / 12 = 0.50

"AAAAAA" -> "BBB"    = 6 /  9 = 0.6

"AAAAAA" -> "AAAAAAC"     = 1 / 13 = 0.08
"AAAAAA" -> "AAAAAACCCC"  = 4 / 16 = 0.25
"AAAAAA" -> "AAAAAACCCCC" = 5 / 17 = 0.29
```
Maybe we can compare the score to `0.25` as for "replacing the half of the string" or similar.

Reply:
I added some tests for different candidate lengths and it seems like @bamx23's algorithm might show better results for some edge cases.
It's now a little bit harder to understand, but I added a comment with a link to this explanation.

Even if we are unable to select the most common route we will fallback to the fastest route which not as bad.